russstation/code/game/machinery/computer/camera_advanced_doors.dm

```diff
@@ -0,0 +1,89 @@
+
+/obj/machinery/computer/camera_advanced/camera_advanced_doors
+	name = "advanced doorknob system"
+	desc = "Used to access the various cameras on the station and remotely open doors."
+	req_access = list()
+
+	light_color = LIGHT_COLOR_RED
+
+/obj/machinery/computer/camera_advanced/camera_advanced_doors/GrantActions(mob/living/user)
+	..()
+	RegisterSignal(user, COMSIG_CAMERA_CLICK_CTRL, .proc/DoorBolt)
+	RegisterSignal(user, COMSIG_CAMERA_CLICK_ALT, .proc/DoorElectrify)
```

Comment:
Could we require the computer to be emmaged to have this electrify proc?

Reply:
That should be doable, but it might be better to make the changes to the door control camera in a separate pr since there are some other changes that I would like to make to it before its ready to use.